Sources/TuistGraph/Models/Target.swift

```diff
@@ -193,93 +214,56 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
     /// Determines if the target is an embeddable watch application
     /// i.e. a product that can be bundled with a host iOS application
     public func isEmbeddableWatchApplication() -> Bool {
-        switch (platform, product) {
-        case (.watchOS, .watch2App), (.watchOS, .app):
-            return true
-        default:
-            return false
-        }
+        let isWatchOS = isExclusiveTo(.watchOS)
+        let isApp = (product == .watch2App || product == .app)
+        return isWatchOS && isApp
     }
 
     /// Determines if the target is an embeddable xpc service
     /// i.e. a product that can be bundled with a host macOS application
     public func isEmbeddableXPCService() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .xpc):
-            return true
-        default:
-            return false
-        }
+        return product == .xpc
     }
 
     /// Determines if the target is an embeddable system extension
     /// i.e. a product that can be bundled with a host macOS application
     public func isEmbeddableSystemExtension() -> Bool {
-        switch (platform, product) {
-        case (.macOS, .systemExtension):
-            return true
-        default:
-            return false
-        }
+        return product == .systemExtension
```

Comment:
for this, if it *is* an extension is defined by the product type, i think we should rely on linting for whether it makes sense for try to define a system extension for a given destination.  I think it would be more informative also for linting to alert the user to this, vs the expected target to just be missing from the generated project.

as for the "can embed" part, it's now possible to have a multi platform app and the `macOS` flavor is the only one that will embed the extension and that will be achieved  with platform filters